### DISCUSSION TITLE
send trigger pulse to pulse stimulator from arduino for optogenetic stimulation

### DISCUSSION THREAD

**Discussion by salvatoreNRIM** [bonsai-rx/bonsai#1685] [Q&A]
Hello everyone!

I have been using the script attached to deliver an optogenetic stimulation through arduino, but the duration of the stimulation fluctuates between 580-650 ms (even though I used 500 ms). Does anyone have any idea why is it so unreliable?
As a possible solution, I decided to try to deliver the optogenetic stimulation by triggering a pulse stimulator (Master9). Do you have any suggestions on how to adapt the code to send the trigger in a reliable way?

[trigger_opto_press_S_Feb2023.zip](https://github.com/bonsai-rx/bonsai/files/14384964/trigger_opto_press_S_Feb2023.zip)

Thank you!

**Comment by bruno-f-cruz** [comment]
Hi
Did you send the correct workflow? All you are doing in the workflow is sending a character 's' to a generic serial port. (?)
